### ISSUE TITLE
[Bug] Switch User

### ISSUE THREAD

**Issue by Ryu945** [obsproject/obs-studio#9618]
### Operating System Info

Other

### Other OS

Nobara 38

### OBS Studio Version

29.1.3

### OBS Studio Version (Other)

RPM Version of OBS

### OBS Studio Log URL

none

### Expected Behavior

When Nobara ( and I suspect this applies to any Fedora based OS) screen savor settings causes the computer to go back to the log in settings.  Then I log in again.  I expect it to not have an affect on the programs that were left running.

### Current Behavior

OBS is somehow primed to disconnect if it was left running while the computer went back to the log in screen.   If the program is not restarted and you try to stream with OBS then a few minutes into the stream it will disconnect with a 100% consistency.  This forced disconnect only occurs once during the stream.

### Steps to Reproduce

1. Start OBS
2. Let the computer go to the log in screen by not using it.
3. Log in
4. Stream with OBS
5. Watch it disconnect a few minutes later
...

### Anything else we should know?

I am using the RPM version of OBS.

**Comment by tytan652** [comment]
Please provide a log or the issue will be closed without asking further information.

**Comment by Ryu945** [comment]
> Please provide a log or the issue will be closed without asking further information.

I tried to generate log when I was not busy with a stream and as irony would have it, something that happens with 100% consistency does not happen when I want it to happen.  I have known about this problem since I had my desktop on Nobara 37 and it is on Nobara 38 now so it is either the major version change fixed it or the window for the forced disconnect has other factors I am not understanding.  I avoided causing this bug for so long that I don't have any log of it happening.  I am going to try to cause this bug during one of my official streams because as luck would have it, that is the only time I seem to get this bug.  Maybe I am doing something in a certain way in my official streams that caused this bug.

Another possibility is that I was using the multi-streaming add-on back then so the forced disconnect could be related to going to log in screen when you use that add-on in the stream.  I managed to go 13 minutes in a test stream and it didn't disconnect.  That is longer then any of the times this bug caused a disconnect.

In short, I'll try causing this bug during my official streams and if the bug mysteriously vanishes, I'll try multistreaming add-on again and see if that is required for the bug to occur.

**Comment by tytan652** [comment]
I asked for a log to see if the functions meant to prevent the session to end/suspend/switch was successfully enabled or not.

Without there is nothing to we can do.

**Comment by gxalpha** [comment]
No log has been provided.

**Comment by Ryu945** [comment]
Problem is not showing  up in Fedora 38.   Either a Fedora update fixed it or and OBS update fixed it.
